### ISSUE TITLE
Get qty of simple products belong to configurable

### ISSUE THREAD

**Issue by marcinex** [owebia/magento2-module-advanced-shipping#86]
Hey guys,
awesome module but i'm stuck with configurable products. Could You please make a small instruction how to get qty of configurable product but connected with customAttribute?

I have two group sizes of paint wall - small and big. Each of them has around 20 specific sizes described by customAttribute. When I want to add to basket a product, it adds me two - one configurable and one simple which is ok. The problem is that simple product holds all customAttributes that I need to search thru and configurable product hold qty of this simple one. I don't know how to get right qty of simple product especially when I have 5 different products in my basket. I need this information to create price rule. I figure out that I should somehow connect item_id with parent_item_id but without success from 2 days. Could You help me with that? I know it is not issues but trying to make it working ...

**Comment by owebia** [comment]
Hi,

You can do something like that to retrieve the parent of an item:

```php
// Define a variable function
$getParent = function ($child) use ($request) {
    $parent = null;
    foreach ($request->all_items as $item) {
        if ($child->parent_item_id == $item->item_id) {
            $parent = $item;
        }
    }
    return $parent;
};

// E.g. of usage
foreach ($request->all_items as $item) {
    $parent = $getParent($item);
    $qty = $parent ? $parent->qty : 0;
}
```

Best Regards,
A.L.

**Comment by marcinex** [comment]
Yes, thank You for that. Works with some small improvements! :+1:

**Comment by wajahatbashir** [comment]
How to we can create shipping method for configurable product ?

I could not understand how to i can implement above logic in this code ?

```
if ($quote->base_subtotal > 364) {

	addMethod('id_0026', [
	'title' 	=> 	"Free Shipping",
        'enabled'   => 	array_sum(
                        array_map(
                            function ($item) {
                            return $item->product->preorder == 0 ? $item->qty : 0;
						},
					$request->all_items
				)
			) == $request->package_qty,
		'price' => 0,
	]);
}
```

**Comment by owebia** [comment]
Duplicate request: https://github.com/owebia/magento2-module-advanced-shipping/issues/110
